core/pkg/core/storage/storage.go

```diff
@@ -0,0 +1,65 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"encoding/json"
+	"database/sql"
+	"mango_truth/core/pkg/core/storage/models"
+	"mango_truth/core/pkg/modules"
+	"github.com/google/uuid"
+	_ "github.com/lib/pq"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+type Storage struct {
+	db boil.ContextExecutor
+}
+
+func NewStorage() *Storage {
+	db, err := sql.Open("postgres", "dbname=mango_truth host=localhost user=zahar password=password sslmode=disable")
+	if err != nil {
+		panic(fmt.Sprintf("Can not connect to the database: %s", err.Error()))
+	}
+	return &Storage{db: db}
+}
+
+func (s *Storage) UpdStatus(status modules.DetectionStatus) {
+	data, err := json.Marshal(status.Verdict)
+	if err != nil {
+		panic(fmt.Sprintf("Can not connect verdict to json in DetectionStatus with requestId=%s, error: %s", status.RequestId, err.Error()))
+	}
+	new_status := models.DetectionStatus{
+		RequestID: status.RequestId.String(),
+		Status: status.Status,
+		Data: null.BytesFrom(data),
+	}
+
+	err = new_status.Insert(context.TODO(), s.db, boil.Infer())
+	if err != nil {
+		panic(fmt.Sprintf("Can not insert value into storage, error: %s", err.Error()))
+	}
+}
+
+func (s *Storage) GetStatus(id uuid.UUID) modules.DetectionStatus {
+	status, err := models.FindDetectionStatus(context.TODO(), s.db, id.String())
+	switch err {
+	case nil:
+		// Do nothing
+	case sql.ErrNoRows:
+		return modules.DetectionStatus{RequestId: id, Status: "UNKNOWN",}
+	default:
+		panic(fmt.Sprintf("Error in FindDetectionStatus, error: %s", err.Error()))
```

Comment:
Don't think, we need to panic. SQL could be down, let's return unknown status and write error to log

Reply:
@ZaraGiraffe

Reply:
I have fixed it. I do not know why it is outdated. See [here](https://github.com/anakib1/MangoTruth/blob/add-sqlboiler-adapter/core/pkg/storage/storage.go)